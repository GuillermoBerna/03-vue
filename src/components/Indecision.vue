<template>
  <img v-if="img" :src="img" alt="bg">
  <div class="bg-dark"></div>
  
  <div class="indecision-container">

    <input v-model="question"
    type="text" 
    placeholder="Hazme Una Pregunta">
    <p>Recuerda colocar un Signo de Interrogacion (?) al final</p>

    <div v-if="isValidQuestion">
      <h2>{{question}}</h2>
      <h1>{{answer}}</h1>
    </div>
  </div>
</template>

<script>
//siempre recordar usar Data y luego establecer las propiedades de los objetos
export default {
  data(){
    return{
      question: null, 
      answer: null,
      img: null,
      isValidQuestion: false
    }
  },
  methods: {
    async getAnswer(){ //se uso Async para crear una sincronizacion con la Api de internet y crear los metodos para mostrarlos en el html
       this.answer = "Pensando..."
       const {image, answer} = await fetch("https://yesno.wtf/api").then(r => r.json())
       this.answer = answer === "yes" ? "Si!" : "No!"
       this.img = image
      }
  },
  //watch sirve para estar pendiente de los cambios que sufre los objetos y decirle que accion tomar cuando estos cambien
  //en este caso se reconoce cuando el usuario coloque un signo de interrogacion con javascript
  watch: {
    question(value, oldValue){

      this.isValidQuestion = false

      console.log({value})

      if( !value.includes('?')) return

      this.isValidQuestion = true

      this.getAnswer()

     }
  }

}
</script>

<style scoped >

    img, .bg-dark {
        height: 100vh;
        left: 0px;
        max-height: 100%;
        max-width: 100%;
        position: fixed;
        top: 0px;
        width: 100vw;
    }

    .bg-dark {
        background-color: rgba(0, 0, 0, 0.4);
    }

    .indecision-container {
        position: relative;
        z-index: 99;
    }
    
    input {
        width: 250px;
        padding: 10px 15px;
        border-radius: 5px;
        border: none;
    }
    input:focus {
        outline: none;
    }

    p {
        color: white;
        font-size: 20px;
        margin-top: 0px;
    }

    h1, h2 {
        color: white;
    }
    
    h2 {
        margin-top: 150px;
    }

</style>