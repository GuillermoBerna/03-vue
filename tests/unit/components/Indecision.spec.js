import {shallowMount} from "@vue/test-utils"
import Indecision from "@/components/Indecision"

describe("indecision component", ()=>{

    let wrapper;

    beforeEach( ()=>{//con beforeEach, se monta el componente antes de la  prueba, luego se crea el test

        wrapper = shallowMount(Indecision)
    })
  
    test("debe de hacer match con el snapshot",()=>{
        expect(wrapper.html()).toMatchSnapshot()

    })










})