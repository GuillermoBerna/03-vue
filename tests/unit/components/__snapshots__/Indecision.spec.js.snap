// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`indecision component debe de hacer match con el snapshot 1`] = `
<!--v-if-->
<div class="bg-dark"></div>
<div class="indecision-container"><input type="text" placeholder="Hazme Una Pregunta">
  <p>Recuerda colocar un Signo de Interrogacion (?) al final</p>
  <!--v-if-->
</div>
`;
